### PR TITLE
Add default linux OS nodeSelector to nmstate handlers

### PR DESCRIPTION
**Pull request by rhrazdil**
This change ensures that nmstate handler pods have the
`kubernetes.io/os: linux` node selector.

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
add kubernetes.io/os=linux nodeSelector to nmstate handler pods by default.
```

### CODE REVIEW

File: pkg/network/nmstate.go [hunk 1]
```diff
@@ -169,3 +175,15 @@ func getMinNumberOfWebhookReplicas(clusterInfo *ClusterInfo) int32 {
 
 	return highlyAvailableMinWebhookReplicas
 }
+
+func pupulateDefaultConfiguration(conf *cnao.NetworkAddonsConfigSpec) *cnao.NetworkAddonsConfigSpec {
```
Comment (qinqon) [inline pkg/network/nmstate.go:179]: This is all at the same package let mark it with "Nmstate"
```suggestion
func pupulateNmstateDefaultConfiguration(conf *cnao.NetworkAddonsConfigSpec) *cnao.NetworkAddonsConfigSpec {
```

**Review by qinqon — CHANGES_REQUESTED** [review]

File: pkg/network/nmstate.go (hunk 1 above)
Comment (rhrazdil) [reply pkg/network/nmstate.go]: Ah, good point. [Updated](https://github.com/kubevirt/cluster-network-addons-operator/compare/d0cd05972a7bccd93e2cea4fce9739e169ae7ee6..e4d4ab762341112ecfdb5f7fb749bd52fa3bc833)

**Pull request merged**